src/main/java/com/rakovets/course/java/core/practice/string/StringUtil.java

```diff
@@ -0,0 +1,209 @@
+        Pattern pattern = Pattern.compile("[aAeEiIoOuU]");
+        Matcher matcher = pattern.matcher(text);
+        while (matcher.find()) {
+            countVowel++;
+        }
+        return countVowel;
+    }
+
+    public int numberOfPunktuationMarks(String text) {
+        int countOfPunctuationMarks = 0;
+        Pattern pattern = Pattern.compile("[.,!?]");
+        Matcher matcher = pattern.matcher(text);
+        while (matcher.find()) {
+            countOfPunctuationMarks++;
+        }
+        return countOfPunctuationMarks;
+    }
+
+    public boolean palindromChekOut(String text) {
+        return text.toLowerCase().replaceAll("[\\pP\\s]", "").equals(new StringBuilder(text.toLowerCase().replaceAll("[\\pP\\s]", "")).reverse().toString());
+    }
+
+    public String[] splitText(String  text, int numberSymbols) {
+        String[] splitText;
+        int j = 0;
+        if (text.length() % numberSymbols != 0) {
+            splitText = new String[text.length() / numberSymbols+1];
+        } else {
+            splitText = new String[text.length() / numberSymbols];
+        }
+
+        for(int i = 0; i < text.length() ; i += numberSymbols) {
+            if (text.length() - i < numberSymbols) {
+                splitText[j] = text.substring(i);
+            } else {
+                splitText[j] = text.substring(i, i + numberSymbols);
+            }
+            j++;
+        }
+
+        return splitText;
+    }
+
+    public int numberOfWords(String text) {
+        int countWords = 0;
+        Pattern pattern = Pattern.compile("[a-zA-Zа-яА-Я]+");
+        Matcher matcher = pattern.matcher(text);
+        while (matcher.find()) {
+            countWords++;
+        }
+
+        return  countWords;
+    }
+
+    public String getFirstLetters(String firstLastName) {
+        String firstLetters = "";
+        String[] container = firstLastName.split("\\s*(\\s)\\s*");
+        for (int i = 0; i < container.length; i ++) {
+            firstLetters += container[i].toUpperCase().charAt(0);
+        }
+
+        return  firstLetters;
+    }
+
+    public String getAllNumbers(String text) {
+        String allNumbers = "";
+        Pattern pattern = Pattern.compile("\\d");
+        Matcher matcher = pattern.matcher(text);
+        while (matcher.find()) {
+            allNumbers += matcher.group();
+        }
+
+        return allNumbers;
+    }
+
+    public String getUniqueSymbols(String firsWord, String secondWord) {
+        String result = "";
+        char[] firstWordChars = firsWord.toCharArray();
+        char[] secondWordChars = secondWord.toCharArray();
+        for (char firstWordChar : firstWordChars) {
+            int count = 0;
+            for (int i = 0; i < secondWordChars.length; i++) {
+                if(firstWordChar == secondWordChars[i]) {
+                    count ++;
+                    secondWordChars[i] = ' ';
+                    break;
+                }
+            }
+            if (count == 0){
```

Comment:
"Violation: There isn't separating white spaces for curly brake
Solution: Add white spaces"